metals/src/main/scala/scala/meta/internal/metals/ScalacDiagnostic.scala

```diff
@@ -1,9 +1,21 @@
 package scala.meta.internal.metals
 
+import scala.jdk.CollectionConverters._
+
 import org.eclipse.{lsp4j => l}
 
 object ScalacDiagnostic {
 
+  object ScalaAction {
+    def unapply(d: l.Diagnostic): Option[l.DiagnosticRelatedInformation] =
+      if (
+        d.getRelatedInformation() != null &&
+        d.getRelatedInformation().asScala.nonEmpty
```

Comment:
If Metals notifies diagnostics with `relatedInformation` (in addition to ActionalDiagnostics stuff) in the future, `ScalaAction` extractor may mis-extract those diagnostics as `ScalaAction`.

I think we can't rely only on the existence of `RelatedInformation` field. We need to check something like an error message or error code or something I guess.